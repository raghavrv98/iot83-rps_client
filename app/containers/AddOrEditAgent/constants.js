/*
 *
 * AddOrEditAgent constants
 *
 */

export const CREATE_AGENT_REQUEST = "app/AddOrEditAgent/CREATE_AGENT_REQUEST";
export const CREATE_AGENT_SUCCESS = "app/AddOrEditAgent/CREATE_AGENT_SUCCESS";
export const CREATE_AGENT_FAILURE = "app/AddOrEditAgent/CREATE_AGENT_FAILURE";

export const GET_AGENT_DETAILS = "app/AddOrEditAgent/GET_AGENT_DETAILS";
export const GET_AGENT_DETAILS_SUCCESS = "app/AddOrEditAgent/GET_AGENT_DETAILS_SUCCESS";
export const GET_AGENT_DETAILS_FAILURE = "app/AddOrEditAgent/GET_AGENT_DETAILS_FAILURE";
