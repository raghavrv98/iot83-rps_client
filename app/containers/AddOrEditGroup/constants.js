/*
 *
 * AddOrEditGroup constants
 *
 */

export const ON_SUBMIT_REQUEST = "app/AddOrEditGroup/ON_SUBMIT_REQUEST";
export const ON_SUBMIT_REQUEST_SUCCESS = "app/AddOrEditGroup/ON_SUBMIT_REQUEST_SUCCESS";
export const ON_SUBMIT_REQUEST_FAILURE = "app/AddOrEditGroup/ON_SUBMIT_REQUEST_FAILURE";

export const GET_GROUP_DETAILS = "app/AddOrEditGroup/GET_GROUP_DETAILS";
export const GET_GROUP_DETAILS_SUCCESS = "app/AddOrEditGroup/GET_GROUP_DETAILS_SUCCESS";
export const GET_GROUP_DETAILS_FAILURE = "app/AddOrEditGroup/GET_GROUP_DETAILS_FAILURE";
