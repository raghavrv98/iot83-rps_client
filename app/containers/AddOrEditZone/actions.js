/*
 *
 * AddOrEditZone actions
 *
 */

import {
  ON_SUBMIT_REQUEST,
  RESET_TO_INITIAL_STATE,
  GET_ZONE_DETAILS
} from "./constants";



export function onSubmitHandler(payload, plantId, pipelineId, zoneId, isBulkUpload) {
  return {
    type: ON_SUBMIT_REQUEST,
    payload,
    plantId,
    pipelineId,
    zoneId,
    isBulkUpload
  };
}
export function getZoneDetails(plantId, pipelineId, zoneId) {
  return {
    type: GET_ZONE_DETAILS,
    plantId,
    pipelineId,
    zoneId
  };
}
export function resetToInitailState() {
  return {
    type: RESET_TO_INITIAL_STATE
  };
}
