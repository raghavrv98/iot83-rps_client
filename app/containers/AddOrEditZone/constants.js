/*
 *
 * AddOrEditZone constants
 *
 */

export const RESET_TO_INITIAL_STATE = "app/AddOrEditZone/RESET_TO_INITIAL_STATE";

export const ON_SUBMIT_REQUEST = "app/AddOrEditZone/ON_SUBMIT_REQUEST";
export const ON_SUBMIT_REQUEST_SUCCESS = "app/AddOrEditZone/ON_SUBMIT_REQUEST_SUCCESS";
export const ON_SUBMIT_REQUEST_FAILURE = "app/AddOrEditZone/ON_SUBMIT_REQUEST_FAILURE";

export const GET_ZONE_DETAILS = "app/AddOrEditZone/GET_ZONE_DETAILS";
export const GET_ZONE_DETAILS_SUCCESS = "app/AddOrEditZone/GET_ZONE_DETAILS_SUCCESS";
export const GET_ZONE_DETAILS_FAILURE = "app/AddOrEditZone/GET_ZONE_DETAILS_FAILURE";
