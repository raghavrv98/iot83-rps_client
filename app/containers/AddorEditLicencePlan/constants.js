/*
 *
 * AddorEditLicencePlan constants
 *
 */

export const DEFAULT_ACTION = "app/AddorEditLicencePlan/DEFAULT_ACTION";

export const ON_SUBMIT_REQUEST = "app/AddorEditLicencePlan/ON_SUBMIT_REQUEST";
export const ON_SUBMIT_REQUEST_SUCCESS = "app/AddorEditLicencePlan/ON_SUBMIT_REQUEST_SUCCESS";
export const ON_SUBMIT_REQUEST_FAILURE = "app/AddorEditLicencePlan/ON_SUBMIT_REQUEST_FAILURE";

export const GET_PLANS_LIST = "app/AddorEditLicencePlan/GET_PLANS_LIST";
export const GET_PLANS_LIST_SUCCESS = "app/AddorEditLicencePlan/GET_PLANS_LIST_SUCCESS";
export const GET_PLANS_LIST_FAILURE = "app/AddorEditLicencePlan/GET_PLANS_LIST_FAILURE";