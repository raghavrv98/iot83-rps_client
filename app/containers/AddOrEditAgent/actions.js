/*
 *
 * AddOrEditAgent actions
 *
 */

import { CREATE_AGENT_REQUEST,GET_AGENT_DETAILS } from "./constants";

export function createAgent(data) {
  return {
    type: CREATE_AGENT_REQUEST,
    payload: data,
  };
}
export function getAgentDetails(id) {
  return {
    type: GET_AGENT_DETAILS,
    id
  };
}
