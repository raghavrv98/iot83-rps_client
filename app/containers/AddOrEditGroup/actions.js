/*
 *
 * AddOrEditGroup actions
 *
 */

import { ON_SUBMIT_REQUEST,GET_GROUP_DETAILS } from "./constants";

export function onSubmitHandler(payload,id) {
  return {
    type: ON_SUBMIT_REQUEST,
    payload,
    id
  };
}

export function getGroupDetails(id) {
  return {
    type: GET_GROUP_DETAILS,
    id
  };
}
