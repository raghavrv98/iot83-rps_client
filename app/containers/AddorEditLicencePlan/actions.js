/*
 *
 * AddorEditLicencePlan actions
 *
 */

import { DEFAULT_ACTION, ON_SUBMIT_REQUEST, GET_PLANS_LIST } from "./constants";

export function defaultAction() {
  return {
    type: DEFAULT_ACTION
  };
}

export function onSubmitHandler(payload) {
  return {
    type: ON_SUBMIT_REQUEST,
    payload
  };
}

export function getPlansList(){
  return{
    type: GET_PLANS_LIST
  }
}