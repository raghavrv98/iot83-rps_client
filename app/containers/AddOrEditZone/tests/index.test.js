import React from 'react';
import { shallow } from 'enzyme';
import { enzymeFind } from 'styled-components/test-utils';
import { onSubmitHandler, getZoneDetails, resetToInitailState } from '../actions';
import { mapDispatchToProps, AddOrEditZone } from '../index';
import { ON_SUBMIT_REQUEST, GET_ZONE_DETAILS, RESET_TO_INITIAL_STATE } from '../constants';
const match = {
	params: {
	  id: "demoId",
	  zoneId: "demo"
	}
  };

const list = {
	isLoading:true
}
describe('< AddOrEditZone />', () => {

	it('match snapshots', () => {
        const wrap = shallow(<AddOrEditZone match={match} onSubmitHandler={onSubmitHandler} resetToInitailState={resetToInitailState} getZoneDetails={getZoneDetails} />)
        expect(wrap).toMatchSnapshot()
    })

	it("test edit case", async () => {
		const resetToInitailState=jest.fn();
		spyOn(AddOrEditZone.prototype, "componentDidMount").and.callThrough();
		const getZoneDetails = jest.fn();
		const wrapper = shallow(<AddOrEditZone match={match} resetToInitailState={resetToInitailState} getZoneDetails={getZoneDetails}/>);
		wrapper.setState({list: list.isLoading})
		expect(AddOrEditZone.prototype.componentDidMount).toHaveBeenCalledTimes(1);
	});

	it('Function testing on mapDispatchToProps', () => {
		const dispatch = jest.fn();
		const payload = 'demo';
		const plantId = 'demo';
		const pipelineId = 'demo';
		const zoneId = 'demo';
		const isBulkUpload = 'demo';
		mapDispatchToProps(dispatch).onSubmitHandler(payload, plantId, pipelineId, zoneId, isBulkUpload);
		expect(dispatch.mock.calls[0][0]).toEqual({
			type: ON_SUBMIT_REQUEST,
			payload: 'demo',
			plantId: 'demo',
			pipelineId: 'demo',
			zoneId: 'demo',
			isBulkUpload: 'demo',
    });
    mapDispatchToProps(dispatch).getZoneDetails(plantId, pipelineId, zoneId);
		expect(dispatch.mock.calls[1][0]).toEqual({
			type: GET_ZONE_DETAILS,
			plantId: 'demo',
			pipelineId: 'demo',
			zoneId: 'demo',
    });
    mapDispatchToProps(dispatch).resetToInitailState();
		expect(dispatch.mock.calls[2][0]).toEqual({
			type: RESET_TO_INITIAL_STATE,
		});
	});
});
