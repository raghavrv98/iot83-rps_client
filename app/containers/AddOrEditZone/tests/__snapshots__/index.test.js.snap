// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`< AddOrEditZone /> match snapshots 1`] = `
<div
  className="appContent"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      AddOrEditZone
    </title>
    <meta
      content="Description of AddOrEditZone"
      name="description"
    />
  </HelmetWrapper>
  <div
    className="pageBreadcrumb"
  >
    <div
      className="flex-item fx-b70"
    >
      <p
        className="pd-l-30"
      >
        <span
          className="cursor-pointer"
          onClick={[Function]}
        >
          <button
            className="btn btn-transparent"
          >
            <i
              className="far fa-long-arrow-left"
            />
          </button>
          <FormattedMessage
            defaultMessage="Manage EHT Zone"
            id="app.common.messages.manageZone"
            values={Object {}}
          >
            <Component />
          </FormattedMessage>
        </span>
      </p>
      <h5>
        <FormattedMessage
          defaultMessage="Update Zone"
          id="app.containers.AddOrEditZone.updateZone"
          values={Object {}}
        >
          <Component />
        </FormattedMessage>
      </h5>
    </div>
  </div>
  <SkeletonLoader
    mode="fullView"
    skeleton="skeletonNestedFormUpload"
  />
</div>
`;
